src/Compilers/Core/Portable/PublicAPI.Unshipped.txt

```diff
@@ -22,9 +22,11 @@ Microsoft.CodeAnalysis.Diagnostics.CompilationWithAnalyzers.GetAnalyzerSyntaxDia
 Microsoft.CodeAnalysis.Diagnostics.CompilationWithAnalyzersOptions
 Microsoft.CodeAnalysis.Diagnostics.CompilationWithAnalyzersOptions.AnalyzerOptions.get -> Microsoft.CodeAnalysis.Diagnostics.AnalyzerOptions
 Microsoft.CodeAnalysis.Diagnostics.CompilationWithAnalyzersOptions.CompilationWithAnalyzersOptions(Microsoft.CodeAnalysis.Diagnostics.AnalyzerOptions options, System.Action<System.Exception, Microsoft.CodeAnalysis.Diagnostics.DiagnosticAnalyzer, Microsoft.CodeAnalysis.Diagnostic> onAnalyzerException, bool concurrentAnalysis, bool logAnalyzerExecutionTime) -> void
+Microsoft.CodeAnalysis.Diagnostics.CompilationWithAnalyzersOptions.CompilationWithAnalyzersOptions(Microsoft.CodeAnalysis.Diagnostics.AnalyzerOptions options, System.Action<System.Exception, Microsoft.CodeAnalysis.Diagnostics.DiagnosticAnalyzer, Microsoft.CodeAnalysis.Diagnostic> onAnalyzerException, bool concurrentAnalysis, bool logAnalyzerExecutionTime, bool reportDiagnosticsWithSourceSuppression) -> void
```

Comment:
These are the new public API needed for this feature:

1. Add a boolean flag _ReportDiagnosticsWithSourceSuppression_ to _CompilationWithAnalyzersOptions_.
2. Add a corresponding constructor for _CompilationWithAnalyzersOptions_.
3. Add a boolean property _Diagnostic.HasSourceSuppression_ to indicate if the diagnostic has a source suppression (pragma or SuppressMessageAttribute).
4. Add a new overload for _Diagnostic.Create_ which takes an additional boolean flag _hasSourceSuppression_. This is needed to allow changing state of a diagnostic from suppressed to active (and vice versa) from the error list.